ee/hogai/eval/eval_insight_search.py

```diff
@@ -0,0 +1,165 @@
+                )
+                for match in id_matches:
+                    try:
+                        insight_id = int(match)
+                        if insight_id not in selected_insight_ids:
+                            selected_insight_ids.append(insight_id)
+                    except ValueError:
+                        pass
+
+    return {
+        "selected_insights": selected_insight_ids,
+        "insights_selected": insights_selected,
+        "creating_new_insight": creating_new_insight,
+        "evaluation_message": evaluation_message,
+        "visualization_count": len(visualization_messages),
+    }
+
+
+@pytest.fixture
+def call_insight_search(demo_org_team_user):
+    """Fixture that creates a callable for executing insight search workflows."""
+    graph = (
+        AssistantGraph(demo_org_team_user[1], demo_org_team_user[2])
+        .add_edge(AssistantNodeName.START, AssistantNodeName.ROOT)
+        .add_root(
+            {
+                "insights": AssistantNodeName.INSIGHTS_SUBGRAPH,
+                "search_documentation": AssistantNodeName.END,
+                "root": AssistantNodeName.END,
+                "end": AssistantNodeName.END,
+                "insights_search": AssistantNodeName.INSIGHTS_SEARCH,
+            }
+        )
+        .add_insights()
+        .add_insights_search()
+        .compile(checkpointer=DjangoCheckpointer())
+    )
+
+    async def callable(search_query: str) -> InsightSearchOutput:
+        conversation = await Conversation.objects.acreate(team=demo_org_team_user[1], user=demo_org_team_user[2])
+        initial_state = AssistantState(
+            messages=[HumanMessage(content=f"Search for insights: {search_query}")],
+            search_insights_query=search_query,
+        )
+
+        raw_state = await graph.ainvoke(initial_state, {"configurable": {"thread_id": conversation.id}})
+        state = AssistantState.model_validate(raw_state)
+
+        eval_info = extract_evaluation_info_from_state(state)
+
+        if eval_info["creating_new_insight"]:
+            evaluation_result = {
+                "should_use_existing": False,
+                "explanation": "Evaluation decided to create new insight (root_tool_insight_plan set)",
+            }
+        elif eval_info["insights_selected"]:
+            evaluation_result = {
+                "should_use_existing": True,
+                "explanation": f"Selected {eval_info['visualization_count']} insight(s) - found visualization messages",
+            }
+        else:
+            evaluation_result = {
+                "should_use_existing": False,
+                "explanation": "No insights selected and no new insight creation triggered",
+            }
+
+        selected_insights = eval_info["selected_insights"]
+        if not selected_insights and eval_info["insights_selected"]:
+            selected_insights = list(range(1, eval_info["visualization_count"] + 1))
```

Comment:
Question: Could you explain this case, please? From what I understand, insight id extraction fails, but `insights_selected` is true, so you create fake insight IDs? Won't it mask the actual issue?